pymc/distributions/timeseries.py

```diff
@@ -179,122 +241,66 @@ def logp(self, value):
     r"""Random Walk with Normal innovations
 
-    Note that this is mainly a user-friendly wrapper to enable an easier specification
-    of GRW. You are not restricted to use only Normal innovations but can use any
-    distribution: just use `aesara.tensor.cumsum()` to create the random walk behavior.
+
+    Notes
+    -----
+    init is currently drawn from a Normal distribution with the same sigma as the innovations
 
     Parameters
     ----------
     mu: tensor
         innovation drift, defaults to 0.0
-        For vector valued mu, first dimension must match shape of the random walk, and
-        the first element will be discarded (since there is no innovation in the first timestep)
     sigma: tensor
-        sigma > 0, innovation standard deviation (only required if tau is not specified)
-        For vector valued sigma, first dimension must match shape of the random walk, and
-        the first element will be discarded (since there is no innovation in the first timestep)
-    tau: tensor
-        tau > 0, innovation precision (only required if sigma is not specified)
-        For vector valued tau, first dimension must match shape of the random walk, and
-        the first element will be discarded (since there is no innovation in the first timestep)
-    init: distribution
-        distribution for initial value (Defaults to Flat())
+        sigma > 0, innovation standard deviation, defaults to 0.0
+    init: float
+        Mean value of initialization, defaults to 0.0
     """
 
-    def __init__(self, tau=None, init=None, sigma=None, mu=0.0, sd=None, *args, **kwargs):
-        kwargs.setdefault("shape", 1)
-        super().__init__(*args, **kwargs)
-        if sum(self.shape) == 0:
-            raise TypeError("GaussianRandomWalk must be supplied a non-zero shape argument!")
-        if sd is not None:
-            sigma = sd
-        tau, sigma = get_tau_sigma(tau=tau, sigma=sigma)
-        self.tau = at.as_tensor_variable(tau)
-        sigma = at.as_tensor_variable(sigma)
-        self.sigma = self.sd = sigma
-        self.mu = at.as_tensor_variable(mu)
-        self.init = init or Flat.dist()
-        self.mean = at.as_tensor_variable(0.0)
-
-    def _mu_and_sigma(self, mu, sigma):
-        """Helper to get mu and sigma if they are high dimensional."""
-        if sigma.ndim > 0:
-            sigma = sigma[1:]
-        if mu.ndim > 0:
-            mu = mu[1:]
-        return mu, sigma
-
-    def logp(self, x):
-        """
-        Calculate log-probability of Gaussian Random Walk distribution at specified value.
+    rv_op = gaussianrandomwalk
+
+    @classmethod
+    def dist(
+        cls,
+        mu: Optional[Union[np.ndarray, float]] = 0.0,
+        sigma: Optional[Union[np.ndarray, float]] = 1.0,
+        init: float = 0.0,
+        size: int = None,
+        *args,
+        **kwargs
+    ) -> RandomVariable:
+
+        return super().dist([mu, sigma, init], **kwargs)
+
+    def logp(
+        value: at.Variable,
```

Comment:
Shouldn't the types be ndarray or TensorVariable?

Reply:
But a numpy ndarray will get used quite a lot here, no?

Reply:
logps are called with the inputs of the respective RandomVariable so they should always be symbolic, even if the user created the distribution with numpy arrays (in make_node they are converted to tensor_variables automatically)

Reply:
Thanks @ricardoV94 :). This is part of the reason I really like that adding type hints, as it helps clear up my confusion as well!